surefire-provider/pom.xml

```diff
@@ -27,6 +27,7 @@
     <dependency>
       <groupId>org.arquillian.smart.testing</groupId>
       <artifactId>smart-testing-core</artifactId>
+      <version>${project.version}</version>
     </dependency>
     <dependency>
       <groupId>org.arquillian.smart.testing</groupId>
```

Comment:
> They should be moved to the functional tests - to have them on classpath. It would be good to have some depchain not to have 4 dependencies but only one...

It's not even about having 1 dep instead of X, but that we can offer all strategies to the users at once without them asking  which dependencies to pull. Good spot  @MatousJobanek, can you create issue here?